packages/wouter/test/link.test-d.tsx

```diff
@@ -56,19 +72,61 @@ describe("Link's ref", () => {
 
     // @ts-expect-error
     <Link to="/" ref={ref}>
-      HELLO
+      Hello
     </Link>;
   });
+});
 
-  it("should work with composed components", () => {
-    const ref = React.useRef<React.ElementRef<typeof Component>>(null);
+describe("<Link /> in asChild mode", () => {
+  it("should work", () => {
+    <Link to="/" asChild>
+      <a>Hello</a>
+    </Link>;
```

Comment:
I would also test custom component just in case to show that not only links are allowed here.

Reply:
typescript can't check children types, so if a regular one works, custom would work too